src/services/utilities.ts

```diff
@@ -419,6 +419,11 @@ namespace ts {
     }
 
     export function positionBelongsToNode(candidate: Node, position: number, sourceFile: SourceFile): boolean {
+        if (candidate.kind === SyntaxKind.IfStatement) {

```

Comment:
I went back and took another look at the smart indenter but I didn't see another place the change could be made. `if` statements are kind of tricky because they can contain gaps (between `if` and `else`), so I though it would be reasonable for `positionBelongsToNode` to encapsulate that knowledge instead of in the smart indenter so that other consumers of this method would also benefit from a more accurate definition of a "position belonging to a node". With that being said, I'm unsure if this method is used anywhere else, so the point may be moot.